File: quivi.pos.react.app/src/hooks/api/Dtos/channels/GetChannelsRequest.ts
import { PagedRequest } from "../PagedRequest";

export interface GetChannelsRequest extends PagedRequest {
    readonly ids?: string[];
    readonly allowsSessionsOnly?: boolean;
    readonly includeDeleted?: boolean;
    readonly search?: string;

    readonly sessionIds?: string[];
    readonly allowsPrePaidOrderingOnly?: boolean;
    readonly allowsPostPaidOrderingOnly?: boolean;
    readonly includePageRanges?: boolean;
    readonly channelProfileId?: string;
    readonly hasOpenSession?: boolean;
}

File: Quivi.Pos.Api/Dtos/Requests/Channels/GetChannelsRequest.cs
namespace Quivi.Pos.Api.Dtos.Requests.Channels
{
    public class GetChannelsRequest : APagedRequest
    {
        public IEnumerable<string>? Ids { get; set; }
        public IEnumerable<string>? SessionIds { get; set; }
        public string? ChannelProfileId { get; set; }
        public string? Search { get; set; }
        public bool AllowsSessionsOnly { get; set; }
        public bool AllowsPrePaidOrderingOnly { get; set; }
        public bool AllowsPostPaidOrderingOnly { get; set; }
        public bool IncludeDeleted { get; set; }
        public bool IncludePageRanges { get; set; }
        public bool? HasOpenSession { get; set; }
    }
}